### ISSUE TITLE
[Bug]: result is not correct for func floor when second para is not valid

### ISSUE THREAD

**Issue by aressu1985** [matrixorigin/matrixone#9034]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):9645bf6e0591daf0e259a0ed613bc1bd94c1d520
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

2023-04-18 11:50:21 ERROR Executor:136 - [/Users/sudong/Program/MatrixOrigin/matrixone/test/distributed/cases/function/func_math_floor.test][row:176][select floor(-123.342124124214, -100);] was executed failed, con[id=0, user=dump, pwd=111].
2023-04-18 11:50:21 ERROR Executor:138 - [EXPECT RESULT]:
floor(-123.342124124214, -100)
-1.0E100
2023-04-18 11:50:21 ERROR Executor:139 - [ACTUAL RESULT]:
floor(-123.342124124214, -100)
-1000000.000000000000

The result of sql(select floor(-123.342124124214, -100);) should be a type of error , not  -1000000.000000000000

**Comment by arjunsk** [comment]
I see that this PR https://github.com/matrixorigin/matrixone/pull/9042/files#diff-f7d19e8a9572d5a606013c064cbfd04343ff8ad3d6c8178bccdb71176c8ae1c5

address this issue.

@aressu1985 , Can I assign this issue to you?

**Comment by arjunsk** [comment]
Hi @zengyan1, Kindly have a look at this issue.

**Comment by zengyan1** [comment]
floor/ceil(decimal,int)当第二个参数过大时报错。
floor/ceil(int/float,int)行为不变

**Comment by aressu1985** [comment]
test today

**Comment by aressu1985** [comment]
fixed: 34c2fa284222e6677d4725a1a6d25919ccf0408f
mysql> select floor(-123.342124124214, -100);
ERROR 1690 (HY000): data out of range: data type decimal64, floor(decimal64(18,12),-100)
mysql>
